### PR TITLE
refactor(module:card): implement new control flow

### PR DESCRIPTION
**Pull request by Nicoss54**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[X] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Application (the showcase website) / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Card component still use structural directive to manage the control flow

Issue Number: N/A

## What is the new behavior?

Card component use the new control flow

## Does this PR introduce a breaking change?
```
[ ] Yes
[X] No
```

### CODE REVIEW

File: components/card/card-loading.component.ts [hunk 1]
```diff
@@ -11,15 +11,15 @@ import { ChangeDetectionStrategy, Component, ViewEncapsulation } from '@angular/
...
-      <div class="ant-row" style="margin-left: -4px; margin-right: -4px;" *ngFor="let listOfClassName of listOfLoading">
-        <div
-          *ngFor="let className of listOfClassName"
-          [ngClass]="className"
-          style="padding-left: 4px; padding-right: 4px;"
-        >
-          <div class="ant-card-loading-block"></div>
+      @for (listOfClassName of listOfLoading; track $index) {
+        <div class="ant-row" style="margin-left: -4px; margin-right: -4px;">
+          @for (className of listOfClassName; track $index) {
```
Comment (HyperLife1119) [inline components/card/card-loading.component.ts:16]: Would it be better to use `track: className`?

Comment (Nicoss54) [reply components/card/card-loading.component.ts:16]: True :) will fix this

Comment (Nicoss54) [reply components/card/card-loading.component.ts]: @HyperLife1119 done :)

File: components/card/card-loading.component.ts [hunk 2]
```diff
@@ -11,15 +11,15 @@ import { ChangeDetectionStrategy, Component, ViewEncapsulation } from '@angular/
...
   template: `
     <div class="ant-card-loading-content">
-      <div class="ant-row" style="margin-left: -4px; margin-right: -4px;" *ngFor="let listOfClassName of listOfLoading">
-        <div
-          *ngFor="let className of listOfClassName"
-          [ngClass]="className"
-          style="padding-left: 4px; padding-right: 4px;"
-        >
-          <div class="ant-card-loading-block"></div>
+      @for (listOfClassName of listOfLoading; track $index) {
```
Comment (HyperLife1119) [inline components/card/card-loading.component.ts:14]: `track： listOfClassName`

**Comment by HyperLife1119** [comment]
It is worth noting that in most cases, choosing a reference as the value of the track is safer than using index, because this is consistent with the previous behavior of ngFor.

File: components/card/card-loading.component.ts (hunk 2 above)
Comment (Nicoss54) [reply components/card/card-loading.component.ts:14]: > `track： listOfClassName`

i'm not sure that we will improve perfromance by tracking on an array but i can make the change

Comment (HyperLife1119) [reply components/card/card-loading.component.ts]: In my opinion, using index as track is not reliable.  In contrast, using an array (object reference) here is more reliable and safe than index.

I hope we can agree on this :)

**Comment by HyperLife1119** [comment]
Test CI failed, can you check it?

**Comment by Nicoss54** [comment]
> Test CI failed, can you check it?

of course

**Comment by Nicoss54** [comment]
@HyperLife1119 all is fixed

**Review by HyperLife1119 — APPROVED** [review]
LGTM

**Pull request merged**